src/Symfony/Component/Mailer/Transport/Dsn.php

```diff
@@ -49,8 +49,8 @@ public static function fromString(string $dsn): self
             throw new InvalidArgumentException(sprintf('The "%s" mailer DSN must contain a host (use "default" by default).', $dsn));
         }
 
-        $user = isset($parsedDsn['user']) ? urldecode($parsedDsn['user']) : null;
-        $password = isset($parsedDsn['pass']) ? urldecode($parsedDsn['pass']) : null;
+        $user = isset($parsedDsn['user']) && '' !== trim($parsedDsn['user']) ? urldecode($parsedDsn['user']) : null;
```

Comment:
no need to trim
`'' !== ($parsedDsn['user'] ?? '') ? urldecode($parsedDsn['user']) : null;`

Reply:
Fixed